cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/DynamicPartitioner.java

```diff
@@ -27,6 +28,27 @@
  */
 public abstract class DynamicPartitioner<K, V> {
 
+  /**
+   * Defines options for when writing to a partition.
+   */
+  public enum PartitionWriteOption {
+    /**
+     * Will only allow writing to a new partition. If an attempt is made to write to a previously-existing partition,
+     * a {@link DataSetException} will be thrown.
+     */
+    CREATE,
+    /**
+     * Allows appending to an existing partition. If the partition does not already exist, it will be created.
+     * Existing partition's metadata is merged with the appending partition's metadata, with the appending partition's
+     * metadata taking precedence if keys conflict.
+     */
+    ALLOW_APPEND,
```

Comment:
Why the ordering matter? (I am ok with that). Also, still, as I said above, I still feel like having the condition separated out and taking a set of options is a better API choice.